file.go

```diff
@@ -293,15 +293,26 @@ func (f *File) reload(s dataSource) error {
 
 // Reload reloads and parses all data sources.
 func (f *File) Reload() (err error) {
-	for _, s := range f.dataSources {
+	for i, s := range f.dataSources {
 		if err = f.reload(s); err != nil {
+			// Notice: if AllowShortCircuit and Loose are both true, AllowShortCircuit has priority over Loose.
+			// Return err when all configuration sources fail to load.
+			if f.options.AllowShortCircuit {
+				if i < len(f.dataSources)-1 {
+					continue
+				}
+				return err
```

Comment:
Why do we return an error here instead of relying on `if os.IsNotExist(err) && f.options.Loose {`?

Reply:
Yeah, you're right, relying on `if os.IsNotExist(err) && f.options.Loose` seems more reasonable.